python_modules/libraries/dagster-dg/dagster_dg/utils/__init__.py

```diff
@@ -259,11 +277,24 @@ def hash_directory_metadata(
             hash_file_metadata(hasher, filepath)
 
 
-def hash_file_metadata(hasher: Hash, path: Union[str, Path]) -> None:
-    stat = os.stat(path=path)
-    hasher.update(str(path).encode())
-    hasher.update(str(stat.st_mtime).encode())  # Last modified time
-    hasher.update(str(stat.st_size).encode())  # File size
+def hash_file_metadata(
+    hasher: Hash, path: Union[str, Path], error_on_missing: bool = False
```

Comment:
suggest defaulting this to True / forcing people to opt in to swallowing exceptions without logging